routes/runtime/github.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+class CloverRuntime
+  hash_branch("github") do |r|
+    if (runner = GithubRunner[vm_id: @vm.id]).nil? || (repository = runner.repository).nil?
+      fail CloverError.new(401, "InvalidCredentials", "There was an error logging in")
+    end
+
+    r.get "cache" do
+      fail CloverError.new(204, "NotFound", "No cache entry") if repository.access_key.nil?
+
+      keys, version = r.params["keys"]&.split(","), r.params["version"]
+      fail CloverError.new(400, "InvalidRequest", "Wrong parameters") if keys.nil? || keys.empty? || version.nil?
+
+      scopes = [runner.workflow_job&.dig("head_branch"), repository.default_branch].compact
+      entry = repository.cache_entries_dataset
+        .exclude(committed_at: nil)
+        .where(key: keys, version: version, scope: scopes).all
+        .min_by { keys.index(_1.key) + (scopes.index(_1.scope) * keys.size) }
```

Comment:
Can you add a comment here? I couldn't really understand how the minimum elements are fetched according to what.

Reply:
I've added some comments there.